engines/terraform/token.go

```diff
@@ -10,4 +20,30 @@ func extractTokenContents(token string) (string, bool) {
 	return "", false
 }
 
+func findAllTokens(input string) []TokenMatch {
+	matches := allTokensPattern.FindAllStringSubmatch(input, -1)
+	indices := allTokensPattern.FindAllStringIndex(input, -1)
+
+	if len(matches) != len(indices) {
+		return nil
+	}
+
+	tokens := make([]TokenMatch, len(matches))
+	for i, match := range matches {
+		tokens[i] = TokenMatch{
+			Token:    match[0],
+			Contents: match[1],
+			Start:    indices[i][0],
+			End:      indices[i][1],
+		}
+	}
+
+	return tokens
+}
+
+func isOnlyToken(input string) bool {
+	return strings.TrimSpace(input) == strings.TrimSpace(allTokensPattern.FindString(input))
+}
+
 var tokenPattern = regexp.MustCompile(`^\${([^}]+)}$`)
+var allTokensPattern = regexp.MustCompile(`\$\{([^}]+)\}`)
```

Comment:
I think this pattern should be far more restricted. Currently it'll still match tokens containing whitespace, tabs, newlines, etc.

<img width="236" height="167" alt="Image" src="https://github.com/user-attachments/assets/d5c47780-4976-4dea-8c74-62008cd3681c" />

Also, @tjholm, not for this PR, but do we want to add an escape char or similar. Are there cases people might need to use values that look like references but have Suga ignore them?